src/hotspot/share/jvmci/jvmciCodeInstaller.cpp

```diff
@@ -923,6 +923,17 @@ int CodeInstaller::estimate_stubs_size(HotSpotCompiledCodeStream* stream, JVMCI_
 // perform data and call relocation on the CodeBuffer
 JVMCI::CodeInstallResult CodeInstaller::initialize_buffer(JVMCIObject compiled_code, CodeBuffer& buffer, HotSpotCompiledCodeStream* stream, u1 code_flags, JVMCI_TRAPS) {
   JavaThread* thread = stream->thread();
+#if defined(__GNUC__) && !defined(__clang__) && !defined(PRODUCT)
+  if (thread == nullptr) {
+    // This is to prevent --stringop-overflow warning from GCC on linux/fastdebug.
+    // GCC does believe that JavaThread::current() can return nullptr,
+    // though it cannot.
+    // GCC seems to not remember that this hint has been given to it when initializing
+    // HotSpotCompiledCodeStream object with a thread returned by JavaThread::current(),
+    // therefore 2nd hint is needed.
+    __builtin_unreachable();
+  }
+#endif
```

Comment:
I think this could be replaced with asserting non-null.  A failed assert calls a noreturn reporting
function.

Reply:
Although I wonder why the assert in Thread::current() isn't sufficient? Maybe non-null info isn't
propagating through the cast to JavaThread?